api_server/src/request/actions.rs

```diff
@@ -176,5 +190,33 @@ mod tests {
                 .unwrap()
                 .eq(&req));
         }
+
+        {
+            let json = r#"{
+                "action_type": "FlushMetrics"
+            }"#;
+
+            let (sender, receiver) = oneshot::channel();
+            let req: ParsedRequest = ParsedRequest::Sync(VmmAction::FlushMetrics(sender), receiver);
+            let result: Result<ActionBody, serde_json::Error> = serde_json::from_str(json);
+            assert!(result.is_ok());
+            assert!(result
+                .unwrap()
+                .into_parsed_request(None, Method::Put)
+                .unwrap()
+                .eq(&req));
+
+            let json = r#"{
+                "action_type": "FlushMetrics",
+                "payload": "metrics-payload"
+            }"#;
+
+            let result: Result<ActionBody, serde_json::Error> = serde_json::from_str(json);
+            assert!(result.is_ok());
+            assert!(result
+                .unwrap()
+                .into_parsed_request(None, Method::Put)
+                .is_err());
```

Comment:
Can you assert on the actual error? You would need to do something like `result.unwrap().into_parsed_request == Err(...)` because `ParsedRequest` does not implement `Debug`.

Reply:
Done. Added a similar check for `test_validate_payload`.